core/systems/input/gamepad_manager.gd

```diff
@@ -310,6 +310,35 @@ func _get_event_from_phys(phys_path: String)  -> String:
 	return event 
 
 
+## Returns true if the InputDevice is a virtual device.
+func _is_device_virtual(device: InputDevice) -> bool:
+	var event := device.get_path()
+	logger.debug("Checking if " + event + " is a virtual device.")
+
+	if not device.get_phys() == "":
+		logger.debug(event + " is real, it has a physical address: " + device.get_phys())
+		return false
+
+	var sysfs_devices := SysfsDevice.get_all()
+	for sysfs_device in sysfs_devices:
+		for handler in sysfs_device.handlers:
```

Comment:
Instead of looping over the handlers, we could do:

```gdscript
var event_file := event.split("/")[-1]  # E.g. "event12"
if not event_file in sysfs_device.handlers:
    continue
...
```